src/tools/linkchecker/main.rs

```diff
@@ -125,16 +142,58 @@ fn main() {
         intra_doc_exceptions: 0,
         has_broken_urls: false,
     };
-    checker.walk(&docs, &mut report);
+    checker.walk(&cli.docs, &mut report);
     report.report();
     if report.errors != 0 {
         println!("found some broken links");
         std::process::exit(1);
     }
 }
 
+fn parse_cli() -> Result<Cli, String> {
+    fn to_canonical_path(arg: &str) -> Result<PathBuf, String> {
+        PathBuf::from(arg).canonicalize().map_err(|e| format!("could not canonicalize {arg}: {e}"))
```

Comment:
For the record, I'm always slightly uneasy with ever using `canonicalize`. I don't think there is anything to change here since it seems to be working. I just wanted to note the risk here.